google/cloud/forseti/notifier/notifiers/inventory_summary.py

```diff
@@ -109,14 +113,29 @@ def _send_email(self, summary_data, details_data):
 
             details_data (list): Details of inventory data as a list of dicts.
                 Example: [[{resource_type, count}, {}, {}, ...]
+
+        Raises:
+            InvalidInputError: Indicating that invalid input was encountered.
         """
         LOGGER.debug('Sending inventory summary by email.')
 
-        email_summary_config = (
-            self.notifier_config.get('inventory').get('email_summary'))
-
-        email_util = email.EmailUtil(
-            email_summary_config.get('sendgrid_api_key'))
+        try:
+            if self.notifier_config.get('email_connector'):
+                email_connector_config_auth = (
+                    self.notifier_config
+                    .get('email_connector')
+                    .get('auth'))
+
+                email_util = SendgridConnector(
```

Comment:
You should not be leaking specific connector here.  The connectors should be gotten from the factory.